localstack/services/secretsmanager/provider.py

```diff
@@ -173,15 +186,24 @@ def delete_secret(
     def describe_secret(
         self, context: RequestContext, request: DescribeSecretRequest
     ) -> DescribeSecretResponse:
-        self._raise_if_invalid_secret_id(request["SecretId"])
-        return call_moto(context, request)
+        secret_id: str = request["SecretId"]
+        self._raise_if_invalid_secret_id(secret_id)
+        backend = SecretsmanagerProvider.get_moto_backend_for_resource(secret_id, context)
+        try:
+            secret = backend.describe_secret(secret_id)
+        except MotoSecretNotFoundException:
+            raise ResourceNotFoundException("Secrets Manager can't find the specified secret.")
+        return DescribeSecretResponse(**secret.to_dict())
 
     @handler("GetResourcePolicy", expand=False)
     def get_resource_policy(
         self, context: RequestContext, request: GetResourcePolicyRequest
     ) -> GetResourcePolicyResponse:
-        self._raise_if_invalid_secret_id(request["SecretId"])
-        return call_moto(context, request)
+        secret_id = request["SecretId"]
+        self._raise_if_invalid_secret_id(secret_id)
+        backend = SecretsmanagerProvider.get_moto_backend_for_resource(secret_id, context)
+        policy = backend.get_resource_policy(secret_id)
+        return GetResourcePolicyResponse(**json.loads(policy))
```

Comment:
Since we are modifying the response generation, have you confirmed that these operations are snapshot tested? I only looked at this op and it appears that the only test using this operation `test_resource_policy` does not use snapshots. Earlier this would be tested in Moto but now we've moved that logic to the provider. I would strongly recommend adding this to make sure we're not introducing any deviations.

Reply:
I have checked all the operations are snapshot validated other than resource policy. I've added snapshot validation for the same in the latest changes.